### ISSUE TITLE
Is beta being trained or is it fixed at zero?

### ISSUE THREAD

**Issue by carlthome** [cooijmanstim/recurrent-batch-normalization#4]
The paper states that both the hidden and input betas are _set_ to the zero vector to avoid unnecessary redundancy (e.g. LSTM bias is enough) but it's not clear if they are trained or just kept constant at zero. They are called _parameters_ throughout, and in the experiments they are _initialized_, implying training.

**Comment by OktayGardener** [comment]
I too thought of what @carlthome is talking about.

**Comment by cooijmanstim** [comment]
Conceptually they are kept at zero; in practice we never create those parameters. If you expand the BN expressions in equation 6, you'll see that you get three terms beta_h + beta_x + b. These parameters are redundant and so we just have one bias/beta vector ("b") instead of three. You can train all three and it probably won't hurt.

**Comment by carlthome** [comment]
Thanks @cooijmanstim! I really appreciate the help. :1st_place_medal: 

Also, I just noticed a mistake in my reproduction (I also set beta_c to zero, derp).
